### PR TITLE
Refactor training and cluster configurations

### PR DESCRIPTION
**Pull request by abdallag**

### CODE REVIEW

File: components/trainer.yaml [hunk 1]
```diff
@@ -16,20 +16,13 @@ name: Train
...
-- {name: GPU Count, type: Integer, default: '1', description: 'Number of VMs in the cluster.'}
-- {name: Model Checkpoint, type: String, description: 'Pretrained model checkpoint.'}
 - {name: Data, type: Dataset, description: 'Preprocessed Data records to train the model on.'}
-- {name: Zone, type: String, description: 'Compute zone to create the cluster VMs in'}
 - {name: Project, type: String, description: 'Project ID where training should run'}
-- {name: Batch Size, type: Integer, description: 'Training batch size'}
-- {name: Epochs, type: Integer, description: 'Training Epochs'}
 - {name: ID, type: String, description: 'Identifier for the training job'}
-- {name: IMAGE_TAG, type: String, description: 'Trainer image tag'}
+- {name: IMAGE, type: String, description: 'Trainer image URL'}
```
Comment (Chris113113) [inline components/trainer.yaml:22]: nit: Image rather than IMAGE?

File: configs/small1vm1gpu.json [hunk 2]
```diff
@@ -3,15 +3,19 @@
...
-  "zone" : "us-central1-a",
-  "node_count" : 1,
-  "model_checkpoint" : "t5-small",
-  "machine_type" : "a2-highgpu-1g",
-  "gpu_type" : "nvidia-tesla-a100",
-  "gpu_count" : 1,
-  "batch_size" : 128,
-  "epochs" : 1,
+  "cluster_config": {
+    "cluster_prefix" : "t5node",
```
Comment (Chris113113) [inline configs/small1vm1gpu.json:7]: Now that it's part of cluster_config `name_prefix` makes more sense than `cluster-prefix`

File: scripts/train/train.sh [hunk 3]
```diff
@@ -13,14 +13,27 @@
...
+
+
+
+gsutil cp ${DATA_DIR}/train_config.json .
+gsutil cp ${DATA_DIR}/cluster.json .
+
+source ./json_to_env.sh ./train_config.json
+source ./json_to_env.sh ./cluster.json
+
+#export MODEL_CHECKPOINT=$1
```
Comment (Chris113113) [inline scripts/train/train.sh:30]: Dead code?

**Review by Chris113113 — APPROVED** [review]

File: components/trainer.yaml (hunk 1 above)
Comment (abdallag) [reply components/trainer.yaml]: Done

File: configs/small1vm1gpu.json (hunk 2 above)
Comment (abdallag) [reply configs/small1vm1gpu.json]: Done

File: scripts/train/train.sh (hunk 3 above)
Comment (abdallag) [reply scripts/train/train.sh]: Done

**Review by Chris113113 — APPROVED** [review]

**Pull request merged**